Work in progress : Tetrahedralization

etienneschmitt opened the pull request:
Je suis en train de rajouter dans cgogn2 plusieurs fonctions de Volume::Modelisation::Tetrahedralization.
Il manque beaucoup de documentation et il n'y a aucun test pour l'instant.
Il y a un ou deux points qui méritent débat. Notamment les fonctions flip_edge et flip_back_edge, qui ne sont que des fonction-outils et que j'ai choisi de ne pas mettre dans CMap3 (ce qui m'oblige à les déclarer fonctions "amis"). Peut-être qu'elles devraient quand même y figurer ?
Comme mon ancienne PR n'est pas mergée il y a des commits qui figurent ici mais qui n'ont pas trop à voir avec la tétrahedrisation.

etienneschmitt committed:
Added bool has_attribute(Orbit, std::string) method in MapBase.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
UINT_MAX -> UINT32_MAX.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
Removed useless "nb_edges_" member in SurfaceImport.
Rewrote SurfaceImport in order to be used directly, without inheritage.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
added classes FileImport, SurfaceFileImport, VolumeFileImport.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
fixed regression in OFF import.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
removed default attribute name in add_attribute.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
New syntaxes for add_attribute and get_attribute.
Added init_attribute.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
using new syntaxes of {add,get}_attribute.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
updated type_traits to handle eigen matrices

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
ChunkArrayGen::import_element(uint32, istream&) : import an element read in an istream.

Signed-off-by: Etienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
move "parse" function in core/utils/serialization.h

Modified the code to be compatible with eigen matrices.

Signed-off-by: Etienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
returning if the vector "table_indices" is empty, avoiding an access to the first element of an empty vec.

Signed-off-by: Etienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
added missing CellMarkerStore::unmark method.

Signed-off-by: Etienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
CMap2::split_vertex{_topo}(std::vector<Dart>&)

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
CMap3::split_vertex CMap3::merge_incident_faces

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
added many functions in modeling/algos/{subdivision,tetrahedralization}.h

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
added missing include.

Signed-off-by: Etienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
added external templates for cgogn::modeling algos.

Signed-off-by: Etienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
added external templates for topology module.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
cgogn geometry external templates.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
fixed some ";;" typos.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
Definitions of AttributeGen, Attribute_T, and Attribute moved to MapBaseData.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
CMap2 flip_back_edge{_topo}

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
CMap3 delete_edge{_topo}

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
ThreadPool marked final. Removed virtual destructor.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
added missing include.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
Added CollectorGen class.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
edge_bisection and swap_gen_32

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
Clarified the doc about the template "phi" method.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
added sew_volumes{_topo}, unsew_volumes{_topo}, delete_volume{_topo}

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
added swap_44 function.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
"CGOGN_USE_EXTERNAL_TEMPLATES" is now defined as a PUBLIC compile option for the cgogn_core target.
Now any external cmake project linking against cgogn_core can benefit from external templates.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
get rid of the FOR_ALL_ORBITS macro.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
again CGOGN_CHECK_CONCRETE_TYPE in flip_edge and flip_back_edge.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
again assert simple_closed_oriented_path in cut_volume_topo.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
fixed merge_incident_faces_topo and some renaming.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
declared modeling::flip_edge and  modeling::flip_back_edge as friend of CMap3_T

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
added flip_edge and flip_back_edge methods.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
CGOGN_NOT_COPYABLE_NOR_MOVABLE(EarTriangulation);

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
fixed compilation.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

pierrekraemer commented:
Les fonctions de flip pourraient effectivement figurer dans CMap3, avec les versions plongées qui vont avec.

etienneschmitt committed:
fixed compilation with MSVC.

Signed-off-by: Etienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
Replaced some macro by template classes.

Removed macros :
check_func_parameter_type
check_func_ith_parameter_type
check_func_return_type
func_parameter_type
func_ith_parameter_type
inside_type

Added classes :
func_arity
func_parameter_type
func_ith_parameter_type
is_func_parameter_same
is_ith_func_parameter_same
func_return_type
is_func_return_same


Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
fixed typo.

Signed-off-by: Etienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
fixed compilation with MSVC2013

Signed-off-by: Etienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
removed "FOR_ALL_ORBITS" macro in cmap2_{quad,tri} and cmap3_{hexa,tetra}

Signed-off-by: Etienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
removed some warnings with MSVC

Signed-off-by: Etienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
Removed all warnings on MSVC 2013

Signed-off-by: Etienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
replaced "#define CGOGN_MODULE_DLL_EXPORT" in each cpp file by target_compile_definitions(${PROJECT_NAME} PRIVATE "-DCGOGN_MODULE_DLL_EXPORT") in the module's CMakeLists.txt

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
Fixed compilation and added external templates for Collector_OneRing and Collector_WithinSphere

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
added missing "selection.cpp" file.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
Make public some VolumeImport methods to construct interactively volume meshes.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
Removed unnecessary nb_vertices_ nb_faces_ and nb_volumes in SurfaceImport and VolumeImport.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
Removed some warnings in the logs.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
Fixed CellMarkerStore and DartMarkerStore

Removed unmark() methods (protected inheritance)
Checking if a dart/cell was already marked before adding it to the vector.

Signed-off-by: Etienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
Fixed wron precondition in CMap3::merge_incident_faces_topo

Signed-off-by: Etienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
Removed some functions in CMap3Builder

Removed phi2_sew, phi2_unsew, add_face_topo.

Signed-off-by: Etienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
Removed VolumeImport::compute_nb_vertices method

Now enforce_unique_orbit_embedding<Vertex::ORBIT>() is always called when creating a map3.

Signed-off-by: Etienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
flip_edge and flip_back_edge

Modified the preconditions to be able to use these functions with a map3.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
Added flip_edge and flip_back_edge in Map3 and removed them from the namespace tetrahedralization.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
Added MapBase::dimension_concrete_map() method.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
Added missing CMap0::DIMENSION.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>
